### ISSUE TITLE
Document @ClientCallable invocations from TS

### ISSUE THREAD

**Issue by OlliTietavainenVaadin** [vaadin/docs#989]
If you have a `@ClientCallable` method and you want to invoke it from TS, you need to either create an explicit type or cast `this` as `any` to be able to invoke the method: `(this as any).$server.methodname()`.

**Comment by tarekoraby** [comment]
@OlliTietavainenVaadin this issue applies both to Flow and Hilla. Right?

**Comment by OlliTietavainenVaadin** [comment]
@tarekoraby no, I don't think there are `@ClientCallable` methods in Hilla, but instead there are Endpoints (and those will have the TS type information available)

**Comment by samie** [comment]
The solution that I found was to declare a private field in my TypeScript element: `private $server: any;`

For example custom HTML element defined in `my-element.ts`: 
```
@customElement('my-element')
export class MyElement extends LitElement {

  private $server: any;

  exampleDomEventHandler(event: Event): void {
    // Call a server-side method
    this.$server.myClientCallableMethod();
  }
  
}
```
And on the server Java class `MyComponent.java`: 
```
@JsModule("./my-element.ts")
public class MyComponent extends Component {

    @ClientCallable
    public void myClientCallableMethod() {
        // We are now on the server side.
    }

}
```

**Comment by OlliTietavainenVaadin** [comment]
You can actually also define a typescript Interface that specifies the type of the client callable method; this is an especially good approach if your server-side method takes arguments or returns a value.
